### PR TITLE
Add unit tests

### PR DESCRIPTION
**Pull request by CM000n**
Proposal for adding unit tests

### CODE REVIEW

**Comment by mawoka-myblock** [comment]
Why did you close this PR? Because there's no way to have a reliable Inverter available, right?

**Comment by CM000n** [comment]
> Why did you close this PR? Because there's no way to have a reliable Inverter available, right?

No, I have closed it for the time being because it looks a bit confusing, as all the changes from the refactoring PR are also displayed.
I will open it again when the refactoring PR has been merged.

**Comment by mawoka-myblock** [comment]
Sure!

**Review by mawoka-myblock — CHANGES_REQUESTED** [review]
Every "raw" response from the inverter is built like the following:
```json
{
	"data": "DATA",
	"status": 0
}
```

If you want, you can add this, since I've never experienced this status being different from `0`.

File: .github/workflows/testing.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      - name: "🐍 Set up Python ${{ matrix.python-version }}"
+        uses: "actions/setup-python@v4"
+        with:
+          python-version: "${{ matrix.python-version }}"
+      - name: "⚙️ Install Poetry"
+        uses: "abatilo/actions-poetry@v2.3.0"
+        with:
+          poetry-version: 1.5.1
+      - name: "⚙️ Install dependencies"
+        run: "poetry install"
```
[the remark this answers is not in the document]
Comment (CM000n) [reply .github/workflows/testing.yml:33]: A `poetry install` will install everything.
You can exclude dev dependencies explicitly with the cli arg `--without dev`

**Comment by CM000n** [comment]
> Every "raw" response from the inverter is built like the following:
> 
> ```json
> {
> 	"data": "DATA",
> 	"status": 0
> }
> ```
> 
> If you want, you can add this, since I've never experienced this status being different from `0`.

Hello, please comment on explicit code line numbers. Such a detached comment makes it somewhat difficult to understand exactly what you mean and where.

**Comment by mawoka-myblock** [comment]
Thanks man! I thought I'd fix these small thinks myself to not annoy you even further! Thank you so much for your work and welcome to 2024 ;) ❤️

**Pull request merged**